library/Icinga/Crypt/RSA.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/* Icinga Web 2 | (c) 2020 Icinga Development Team | GPLv2+ */
+
+namespace Icinga\Crypt;
+
+use InvalidArgumentException;
+use UnexpectedValueException;
+
+/**
+ * Class RSA
+ *
+ * This class allows to encrypt and decrypt data.
+ *
+ * Usage
+ * -----
+ * <code>
+ * ** Encryption **
+ *
+ * $rsa = new RSA();
+ * $rsa->loadKey(...RSA::keygen());
+ *
+ * just Encrypt data
+ * $data = $rsa->encrypt($myVar1, myVar2); // multiple parameters allowed
+ *
+ * Encrypt and encodes the given data with MIME base64
+ * $data = $rsa->encryptToBase64($myVar1, myVar2); // multiple parameters allowed
+ *
+ *
+ * ** Decryption **
+ *
+ * $rsa = new RSA();
+ * $rsa->loadKey($publicKey, load_private_key_from_database($publicKey));
+ *
+ * Decrypt and decodes the given data which is encoded in MIME base64
+ * list($myVar1, $myVar2) = $rsa->decryptFromBase64(...$data);
+ * </code>
+ *
+ * @package Icinga\Crypt
+ */
+class RSA
+{
+    /** @var string */
+    private $pubKey;
+
+    /** @var string */
+    private $privKey;
+
+    /**
+     * Generates a new private and public key pair
+     *
+     * @return array
+     */
+    public static function keygen()
+    {
+        $res = openssl_pkey_new(
```

Comment:
Add the array brackets on the same lines as the function opening and closing parenthesis, i.e.

```php
foobar([
    ...
]);
```

This is ok here since the array is the one and only parameter.